### PR TITLE
[quant] Fix fusion pattern for add_relu

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39380 [quant][graphmode] Test for another type of ops in insert_observer for if
* #39368 [quant][graphmode][fix] Run preprocess for child module before parent module
* **#39367 [quant] Fix fusion pattern for add_relu**

Summary:
We shouldn't match `%alpha` argument since it could be used by multiple functions

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21829295](https://our.internmc.facebook.com/intern/diff/D21829295)

### CODE REVIEW

File: torch/csrc/jit/passes/quantization/quantization_patterns.h [hunk 1]
```diff
@@ -241,8 +241,7 @@ graph(%a_quant, %packed_params, %r_scale, %r_zero_point, %r_dtype, %stride, %pad
         return (%r_quant) )";
 
   std::string add_relu = R"(
-graph(%a_quant, %b_quant, %scale, %zero_point, %dtype):
-         %alpha = prim::Constant[value=1]()
+graph(%a_quant, %b_quant, %alpah, %scale, %zero_point, %dtype):
```
Comment (supriyar) [inline torch/csrc/jit/passes/quantization/quantization_patterns.h:244]: should be `alpha`. Do we have test to catch this?

File: torch/csrc/jit/passes/quantization/quantization_patterns.h [hunk 2]
```diff
@@ -281,7 +277,7 @@ graph(%a_quant, %b_quant, %scale, %zero_point, %dtype):
          return (%r) )";
 
   std::string quantized_add_relu = R"(
-graph(%a_quant, %b_quant, %scale, %zero_point, %dtype):
+graph(%a_quant, %b_quant, %alpha,  %scale, %zero_point, %dtype):
```
Comment (supriyar) [inline torch/csrc/jit/passes/quantization/quantization_patterns.h:280]: nit, extra space?

**Comment by supriyar** [comment]
Should we add or update existing test to catch this in the future?

File: torch/csrc/jit/passes/quantization/quantization_patterns.h (hunk 1 above)
Comment (jerryzh168) [reply torch/csrc/jit/passes/quantization/quantization_patterns.h:244]: yeah, the tests were failing and I was trying to figure out why...

**Comment by jerryzh168** [comment]
> Should we add or update existing test to catch this in the future?

yeah we have test for this, I forgot to run the test before submit, please take a look again.

**Comment by supriyar** [comment]
> > Should we add or update existing test to catch this in the future?
> 
> yeah we have test for this, I forgot to run the test before submit, please take a look again.

I was referring to testing the case where `alpha` is used in multiple places.

**Comment by jerryzh168** [comment]
> I was referring to testing the case where `alpha` is used in multiple places.

Oh, I see, yeah I can add a test for this

**Review by supriyar — APPROVED** [review]

**Pull request closed without merging**